services/evm/executor.go

```diff
@@ -0,0 +1,196 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/onflow/atree"
+	"github.com/onflow/flow-go/fvm/evm"
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/onflow/flow-go/fvm/evm/emulator/state"
+	"github.com/onflow/flow-go/fvm/evm/precompiles"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	flowGo "github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/go-ethereum/common"
+	gethTypes "github.com/onflow/go-ethereum/core/types"
+	"github.com/onflow/go-ethereum/eth/tracers"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-evm-gateway/models"
+	"github.com/onflow/flow-evm-gateway/storage"
+)
+
+type BlockExecutor struct {
+	types.StateDB // todo change to types.ReadOnlyView
+	emulator      types.Emulator
+	chainID       flowGo.ChainID
```

Comment:
you could store the EVM chain ID since that is whats needed further down.

Reply:
Good idea :+1: 
Updated in https://github.com/onflow/flow-evm-gateway/pull/605/commits/8417e1f26d31d0a6c909c8d57c2df3bb53bb467f